app/models/order_detail.rb

```diff
@@ -845,16 +846,18 @@ def manual_fulfilled_at=(string)
   end
 
   def time_data
-    @time_data ||= product.time_data_for(self)
+    if product.respond_to?(:time_data_for)
```

Comment:
This was the solution I came up with for dealing with items/services should be treated as null objects, but instruments require a reservation, so `product.time_data_for(self) || NullTimeData.new` was problematic in the cases where the instrument order doesn't have a reservation (mostly happens in the cart).